crates/lld-sys/build.rs

```diff
@@ -22,27 +39,74 @@ fn set_rustc_link_flags() {
         "LLVMTargetParser",
         "LLVMBinaryFormat",
         "LLVMDemangle",
+        // Required by `llvm-sys`
```

Comment:
The llvm-sys crate uses a call to llvm-config to get the list of libraries to link against. However, when cross-compiling, you encounter an issue as described in this GitHub issue: https://github.com/llvm/llvm-project/issues/58984. Therefore, I had to manually provide all the libraries."

Reply:
But the build of the compiler itself is never executed in emscripten? So this means for some reason in the emscripten target `llvm-config` becomes a dependency? I'll need to look into it as I'd like to understand whats going on here exactly.

Reply:
Thanks. So if I understand that right, `llvm-sys` is a runtime dependency _during buildtime_. And in `emscripten`, instead of shelling out it directly runs it. That would make sense to me.

Can you please expand the comment a bit? Because otherwise it isn't obvious why all those are needed :)

Reply:
`llvm-sys` is a crate which has buildtime dependency on `llvm-config` binary.
Comment expanded